server/lib/tuist/qa/agent.ex

```diff
@@ -15,6 +15,16 @@ defmodule Tuist.QA.Agent do
 
   require Logger
 
+  defp log_and_stream(message, log_streamer, level \\ :debug) do
+    Logger.log(level, message)
+
+    Client.stream_log(log_streamer, %{
+      message: message,
+      level: to_string(level),
+      timestamp: DateTime.utc_now()
+    })
+  end
```

Comment:
`Logger` supports [backends](https://hexdocs.pm/logger/1.12.3/Logger.html#module-backends). What if you implement a module, `Tuist.QA.Logger.ServerBackend` and then configure the `Logger` with:

```exs
Logger.add_backend(:console)  
Logger.add_backend(Tuist.QA.Logger.ServerBackend)  
```

Reply:
Perhaps this only makes sense once the agent runs as a standalone Elixir application with its own logger configuration.